### PR TITLE
Stop handoff if segment has been compacted

### PR DESCRIPTION
**Pull request by xige-16**
fix issue: #15139 
/kind bug
Signed-off-by: xige-16 <xi.ge@zilliz.com>

### CODE REVIEW

File: internal/querycoord/index_checker.go [hunk 1]
```diff
@@ -192,13 +193,20 @@ func (ic *IndexChecker) checkIndexLoop() {
 				validHandoffReq, collectionInfo := ic.verifyHandoffReqValid(segmentInfo)
 				if validHandoffReq && Params.QueryCoordCfg.AutoHandoff {
 					indexInfo, err := getIndexInfo(ic.ctx, segmentInfo, collectionInfo.Schema, ic.rootCoord, ic.indexCoord)
-					if err != nil {
+					if err == nil {
+						// if index exist or not enableIndex, ready to load
+						segmentInfo.IndexInfos = indexInfo
+						ic.enqueueIndexedSegment(segmentInfo)
+						break
+					}
```
Comment (congqixia) [inline internal/querycoord/index_checker.go:201]: If feels weird to put normal branch in if block while left the error route below

File: internal/querycoord/index_checker.go [hunk 2]
```diff
@@ -192,13 +193,20 @@ func (ic *IndexChecker) checkIndexLoop() {
...
+
+					// if segment has not been compacted and dropped, continue to wait for the build index to complete
+					segmentState, err := getSegmentStates(ic.ctx, segmentInfo.SegmentID, ic.dataCoord)
+					if err != nil || segmentState.State != commonpb.SegmentState_NotExist {
```
[the remark this answers is not in the document]
Comment (xige-16) [reply internal/querycoord/index_checker.go:205]: > 

there will be a lot of logs printed before the index is built.

File: internal/querycoord/index_checker.go (hunk 1 above)
Comment (xige-16) [reply internal/querycoord/index_checker.go:201]: > If feels weird to put normal branch in if block while left the error route below

The error just means that the index has not been built, not that there is an error when running.

File: internal/querycoord/index_checker.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (congqixia) [reply internal/querycoord/index_checker.go:205]: The error is not nil when grpc call fails so it's fine to print the error here

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**